src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.unit.rest;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import org.glassfish.jersey.internal.util.collection.ImmutableMultivaluedMap;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.baidu.hugegraph.rest.RestClient;
+import com.baidu.hugegraph.rest.RestResult;
+import com.baidu.hugegraph.testutil.Assert;
+import com.google.common.collect.ImmutableMap;
+
+public class RestClientTest {
+
+    private static RestClient restClient = new RestClientImpl("/test", 3000);
+
+    private static class RestClientImpl extends RestClient {
+
+        public RestClientImpl(String url, int timeout) {
+            super(url, timeout);
+        }
+
+        @Override
+        protected Response request(Callable<Response> method) {
+            Response response = Mockito.mock(Response.class);
+            Mockito.when(response.getStatus()).thenReturn(200);
+            Mockito.when(response.getHeaders())
+                   .thenReturn(ImmutableMultivaluedMap.empty());
+            Mockito.when(response.readEntity(String.class))
+                   .thenReturn("");
+            return response;
+        }
```

Comment:
please mock javax.ws.rs.client.SyncInvoker.post/put/get, and check path and parameter is expected, return a response and assert RestResult is expected.